### PR TITLE
caddy: add new project

### PR DESCRIPTION
**Pull request by Mohammed90**
This PR adds Caddy. I've tested the setup locally using the guide described in the [Setting up new project#Testing Locally](https://google.github.io/oss-fuzz/getting-started/new-project-guide/#testing-locally) page.

The coverage sanitizer doesn't seem to work nicely with Go's SIV, so we opted to not support it for now. The fix will either have to be in the shell script used for the coverage sanitizer or we figure out a hack in our `build.sh`.

### CODE REVIEW

**Comment by DavidKorczynski** [comment]
I think this is a duplicate of over here: https://github.com/google/oss-fuzz/pull/5370 
Can we merge the fuzzers over there upstream and then get the PRs in @AdamKorcz

File: projects/caddy/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Copyright 2020 Google Inc.
```
Comment (AdamKorcz) [inline projects/caddy/Dockerfile:1]: nit: Should be `2021 Google LLC` instead of `2020 Google Inc.`

File: projects/caddy/build.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash -eu
+# Copyright 2020 Google Inc.
```
Comment (AdamKorcz) [inline projects/caddy/build.sh:2]: nit: Should be `2021 Google LLC` instead of `2020 Google Inc.`

**Comment by Mohammed90** [comment]
> I think this is a duplicate of over here: #5370 - I think it makes sense to just stick to that one.
> Can we merge the fuzzers from over there onto upstream, and then consider it for integration @AdamKorcz 

I've added my feedback on #5370. We can continue there.

> Who are the customers of caddy, i.e. which projects use it?

From the top of my head:
- [ohdear](https://ohdear.app/blog/how-we-used-caddy-and-laravels-subdomain-routing-to-serve-our-status-pages)
- [Approximated](https://approximated.app/)
- [Mercure](https://mercure.rocks/)
- [API-Platform](https://api-platform.com/)
- [Fathom Analytics](https://laravel-news.com/unlimited-custom-domains-vapor)

**Comment by Mohammed90** [comment]
Thank you to @AdamKorcz for writing the initial setup!

This PR is ready.

**Comment by AdamKorcz** [comment]
The previous PR has been closed. We will move forward with this one.

**Comment by jonathanmetzman** [comment]
@Mohammed90 Sorry for the delay in reviewing this.
Internally I started our standard review on whether caddy is a good project for OSS-Fuzz.
I think we will probably be able to finish this (and possibly merge this PR) by today.

**Review by jonathanmetzman — APPROVED** [review]
LGTM

**Pull request merged**